### PR TITLE
[RFC] Create Accents based on user input

### PR DESCRIPTION
**Pull request by timunie**
**Describe the changes you have made to improve this project**

The ThemeManager has a got a new function that takes a userdefined color and creates a new Accent from it. The Accent will be created for all base themes, so you can switch to light and dark as usual.

Please let me know, what you think about this :-)

You can test it in the demo app under accents-menu.

**Unit test**

I don't know how to do, but if anybody can give me a hint i will do.

**Additional context**

image

image

image

**Closed Issues**

### CODE REVIEW

File: src/MahApps.Metro.Samples/MahApps.Metro.Demo/ExampleViews/ColorExample.xaml.cs [hunk 1]
```diff
@@ -1,4 +1,8 @@
-﻿using System.Windows.Controls;
+﻿using MahApps.Metro;
+using System.Linq;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Media;
```
Comment (punker76) [inline src/MahApps.Metro.Samples/MahApps.Metro.Demo/ExampleViews/ColorExample.xaml.cs]: Please revert these change by adding the other uses.

File: src/MahApps.Metro/ThemeManager/ThemeManagerHelper.cs [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Markup;
+using System.Windows.Media;
+using generator = XamlColorSchemeGenerator;
```
[the remark this answers is not in the document]
Comment (timunie) [reply src/MahApps.Metro/ThemeManager/ThemeManagerHelper.cs]: @punker76 i understand you regarding the depencies, that is something i want to reduce as much as possible, too. Do we have the depency to the JSON-Converter? If yes, i think all we need to change is to replace the {{colortags}} with the color name. I could make a suggestion in the next days, if you want me to.

Another question: Some Codacy checks were not ok. But i don't know how to solve them. What do I have to do here? 


Thank you very much for your support!

**Comment by batzen** [comment]
I already ported the generator tool to .net core 3.0 on my machine, but i had to add a dependency to newtonsoft.json as there was, at that point in time, no json serializer available in .net core 3.0 itself.
Preview 5 added a serializer to System.Text.Json.
I will update my local branch to use that for .net core and the old javascript json serializer for .net 4.5.
After i have done that i could move all files/classes to a single generator file.

Why don't you want to add another dependency to MahApps.Metro?
I could try to share the code from the generator repository with ControlzEx, but i don't think that's a good idea.

**Comment by batzen** [comment]
Thinking further about this it seems wiser to add the generator to ControlzEx somehow as adding the generator as a new dependency to MahApps.Metro would also require the same dependency in Fluent.Ribbon in the future.

**Comment by punker76** [comment]
> Thinking further about this it seems wiser to add the generator to ControlzEx somehow as adding the generator as a new dependency to MahApps.Metro would also require the same dependency in Fluent.Ribbon in the future.

That was my thought, too.

File: src/MahApps.Metro.Samples/MahApps.Metro.Demo/MahApps.Metro.Demo.csproj [hunk 3]
```diff
@@ -31,18 +31,19 @@
...
-        <Reference Include="WindowsFormsIntegration"/>
+        <Reference Include="System.Windows.Forms" />
+        <Reference Include="WindowsFormsIntegration" />
     </ItemGroup>
     <ItemGroup>
-        <None Include="app.manifest"/>
-        <None Remove="App.config"/>
+        <None Include="app.manifest" />
+        <None Remove="App.config" />
+        <None Remove="ExampleViews\AddCustomAccentDialog.xaml" />
```
[the remark this answers is not in the document]
Comment (timunie) [reply src/MahApps.Metro.Samples/MahApps.Metro.Demo/MahApps.Metro.Demo.csproj]: I don't know why this happend, i didn't change this file. Visual Studio must have done it. I used the SDKs mentioned by the help file: 

[https://github.com/MahApps/MahApps.Metro/wiki/Building-the-MahApps.Metro-solution](url)

File: src/MahApps.Metro.Samples/MahApps.Metro.Demo/MainWindow.xaml [hunk 4]
```diff
@@ -9,12 +9,10 @@
                       xmlns:iconPacks="http://metro.mahapps.com/winfx/xaml/iconpacks"
                       xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
                       Title="MahApps.Metro - Demo Application"
-                      Width="900"
-                      Height="600"
+                      Width="1352.011"
+                      Height="620.912"
```
[the remark this answers is not in the document]
Comment (timunie) [reply src/MahApps.Metro.Samples/MahApps.Metro.Demo/MainWindow.xaml]: Oh sorry, maybe i hit the scaling thumb in the design view. This was not my intention.

**Comment by timunie** [comment]
Great idea to make this PR public for other projects by moving it to ControlzEx. Maybe it can be used for the MaterialDesign, too. 

@batzen : For now i will stop working on this PR, because i think it would be the same work twice. :heavy_check_mark: If I can support anyway, please contact me. 
@punker76 : Can you set this PR on hold or close it or move it to ControlzEx?

**Comment by timunie** [comment]
I close this PR because we have this feature now thanks to @batzen
